tests/components/risco/util.py

```diff
@@ -17,30 +17,77 @@
 TEST_SITE_NAME = "test-site-name"
 
 
-async def setup_risco(hass, events=[], options={}):
+class MockRiscoCloud:
```

Comment:
Consider patching the RiscoAPI class instead and autospeccing it. Ie don't patch `get_risco_cloud` but patch the class that it instantiates. This may make it easier to keep the tests in sync with the library.

Reply:
Wouldn't that require importing the class which would increase the dependency on implementation? Currently it's not imported anywhere

Reply:
We can patch the name in the library, I think.

Reply:
OK, so for example:
```python
with patch("pyrisco.cloud.RiscoAPI.login", side_effect=exception):
```
?
That would bring it close to how the code was before this PR, but I feel like it requires more intimate knowledge of the library, and will break when classes move around.

Reply:
Yes. It doesn't require more knowledge than the library interface.

`get_risco_cloud` is just a proxy for `RiscoAPI` and not really needed. I would consider removing those proxy functions in favor of instantiating the classes directly. The library user interacts with these classes anyway.

Reply:
ok, done. The diff is significantly smaller now